os_net_config/schema.yaml

```diff
@@ -15,6 +15,8 @@ definitions:
             get_input:
                 type: string
           additionalProperties: False
+        - type: string
+          pattern: "^self$"
```

Comment:
I have reservation on changing the base param - because base param will be used by:
- dns_server ( "ip_address_string_or_param" )
- ip_cidr_string_or_param
- route_rule
Would this change make "self" being allowed for all of the above ?

Maybe a new custom type is safer

Reply:
every $ref: "#/definitions/param call will allow "self" (as is) string.
do you see any risk here ?

Reply:
There is no risk with accepting self as-is, the danger is if we open this up in a follow-up to accept interface names, but I think it's acceptable, after all the user running this as root is also the one supplying the input template, so they accept the risk themselves.

Reply:
Viji is correct about not changing this base type but rather creating a custom type, we don't want to limit route rules or dns server to only being "self", so we want to have a custom type here that applies to the route object but not route_rule or dns_server, etc.

Reply:
I think we should revert to allowing all strings for now, as I mentioned the risk is the same as we had before, and the user running os-net-config is providing the input so they are accepting the risk. Perhaps we can come up with a more restrictive string that doesn't allow escape characters, but I think that the oslo process execution module we are using already has some input sanitizing, so I think we are covered. Let's just allow any string for now and mark this for review in the near future.

Reply:
OK
added patchset (to avoid scheme errors) with new type "self_interface" with self string for now ... you can expend the regex to accept interface names later on for